tools/timer/main.c

```diff
@@ -25,7 +25,7 @@ int main()
 	fprintf(stderr, "gettimeofday error");
 	return 1;
     }
-    fprintf(stdout, "%li.%li\n", t.tv_sec, t.tv_usec);
+    fprintf(stdout, "%li.%li\n", t.tv_sec, (long)t.tv_usec);
```

Comment:
`long` is not portable, but platform-dependent, might be a 32bit integer or a 64bit integer. Use `grass_int64` if you need a portable 64bit integer.

Reply:
"The type suseconds_t shall be a signed integer type capable of storing values at least in the range [-1, 1000000]." [IEEE](https://pubs.opengroup.org/onlinepubs/007904975/basedefs/sys/types.h.html#tag_13_67), which means a 32 bit integer will be enough. Apparently [GNU C](https://www.gnu.org/software/libc/manual/html_node/Time-Types.html) and [FreeBSD](https://svnweb.freebsd.org/base/head/sys/sys/_types.h?view=markup) among others choose to use `long int`, whereas on darwin (mac) it is actually an int32. The easiest and most common solution for `printf()` is to upcast to `long` as I suggested.